unstructured/ingest/connector/pinecone.py

```diff
@@ -0,0 +1,144 @@
+import json
+import multiprocessing as mp
+import typing as t
+import uuid
+from dataclasses import dataclass
+
+from unstructured.ingest.error import DestinationConnectionError, WriteError
+from unstructured.ingest.interfaces import (
+    BaseConnectorConfig,
+    BaseDestinationConnector,
+    BaseIngestDoc,
+    BaseSessionHandle,
+    ConfigSessionHandleMixin,
+    IngestDocSessionHandleMixin,
+    WriteConfig,
+)
+from unstructured.ingest.logger import logger
+from unstructured.staging.base import flatten_dict
+from unstructured.utils import requires_dependencies
+
+if t.TYPE_CHECKING:
+    from pinecone import Index as PineconeIndex
+
+
+@dataclass
+class PineconeSessionHandle(BaseSessionHandle):
+    service: "PineconeIndex"
+
+
+@DestinationConnectionError.wrap
+@requires_dependencies(["pinecone"], extras="pinecone")
+def create_pinecone_object(api_key, index_name, environment):
+    import pinecone
+
+    pinecone.init(api_key=api_key, environment=environment)
+    index = pinecone.Index(index_name)
+    logger.debug(f"Connected to index: {pinecone.describe_index(index_name)}")
+    return index
+
+
+@dataclass
+class SimplePineconeConfig(ConfigSessionHandleMixin, BaseConnectorConfig):
+    api_key: str
+    index_name: str
+    environment: str
+
+    def create_session_handle(self) -> PineconeSessionHandle:
+        service = create_pinecone_object(self.api_key, self.index_name, self.environment)
+        return PineconeSessionHandle(service=service)
+
+
+@dataclass
+class PineconeWriteConfig(IngestDocSessionHandleMixin, WriteConfig):
+    connector_config: SimplePineconeConfig
+    batch_size: int = 50
+    num_processes: int = 1
+
+
+@dataclass
+class PineconeDestinationConnector(BaseDestinationConnector):
+    write_config: PineconeWriteConfig
+    connector_config: SimplePineconeConfig
+
+    def initialize(self):
+        pass
+
+    def check_connection(self):
```

Comment:
If this fails, it should raise the custom `SourceConnectionError` error.

Reply:
This is already ensured with this part (since `check_connection` calls `create_pinecone_object`):
https://github.com/Unstructured-IO/unstructured/blob/7e0c7e7a97237679ccd4d5a1c34385a523eb001d/unstructured/ingest/connector/pinecone.py#L30-L32

However, I've added an additional wrapper with [an additional error wrapper for check connection](https://github.com/Unstructured-IO/unstructured/pull/1774/commits/b54e5ce3ed31915a6dd8f69e95034e9b925c307a) in case check_connection gets modified (and does not call create_pinecone_object) in the future